### ISSUE TITLE
Missing CS8908

### ISSUE THREAD

**Issue by TheLeftExit** [dotnet/roslyn#58878]
**Version Used**: 
Microsoft Visual Studio Community 2022 (64-bit) Version 17.0.2
.NET 6,

**Steps to Reproduce**:

1. Create a project using the .NET 6 Console App template
2. Add the following line at the end of `Program.cs`:
```cs
public unsafe record MyStruct(void* Value1, nuint Value2);
```
3. Enable unsafe code in the project using the `Quick Actions and Refactorings...` context menu

**Expected Behavior**:
CS8908 is shown in Error List, the line is marked.

**Actual Behavior**:
No errors are shown in the editor before and after attempted build. CS8908 is only shown in the Output window:
```plain-text
CS8908: The type 'void*' may not be used for a field of a record.
```

**Comment by jcouv** [comment]
I suspect it's because this error (`ERR_BadFieldTypeInRecord`) is generated during the emit phase of the compilation. The IDE and Error List don't report those well.
